internal/pkg/dl/migration.go

```diff
@@ -61,11 +61,16 @@ func migrate(ctx context.Context, bulker bulk.Bulk, fn migrationBodyFn) (int, er
 	for {
 		name, index, body, err := fn()
 		if err != nil {
-			return updatedDocs, fmt.Errorf(": %w", err)
+			return updatedDocs,
+				fmt.Errorf("failed to prepare request for migration %s: %w",
+					name, err)
 		}
 
 		resp, err := applyMigration(ctx, name, index, bulker, body)
 		if err != nil {
+			log.Err(err).
+				Bytes("http.request.body.content", body).
```

Comment:
Are you concerned about how large body could be here? Or that it might contain information we want to sanitize out of the logs? I am not sure what exactly will be logged here so perhaps this isn't a concern.

Besides this small concern it looks good.

Reply:
I'm a bit concerned about the size, but I think it'd be fine. The largest it might log is that [script/request body](https://github.com/elastic/fleet-server/blob/16ee12afe3ba8740b3bfa568d079a493e7664b9a/internal/pkg/dl/migration.go#L230-L264)

Reply:
worst case, AFAIK, ES wont take the document in or truncate it, so not a big issue